src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,255 @@
+package mate.jdbc.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import mate.jdbc.exception.DataProcessingException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.lib.Inject;
+import mate.jdbc.model.Car;
+import mate.jdbc.model.Driver;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class CarDaoImpl implements CarDao {
+    @Inject
+    private DriverDao driverDao;
+    @Inject
+    private ManufacturerDao manufacturerDao;
```

Comment:
in my opinion it will be better to not use here driverDao and manufacturerDao.
isAllDataOk method could just throw exceptions if you try to add car with not existing manufacturer or drivers. Not create new driver or manufacturer. But is only my opinion. You could leave this code as you think better.

Reply:
I want to try with them!

Reply:
probably you are right!